### ISSUE TITLE
Error: StaticInjectorError[Injector]: when consuming library

### ISSUE THREAD

**Issue by Howisch92** [dherges/ng-packagr#764]
## Type of Issue

```
[?] Bug Report
[ ] Feature Request
[x] Help Rquest
```

## Description
Relatively simple error, but i haven't found any solution that works in my case. 
i try to package a component library that i created, and everything works fine until i try to include a service which uses Httpclient for get requests.

i've seen several dusin solutions, including --preserveSymlinks, a bunch of suggestions, 
on where and what to import / DI (HttpModule --> HttpClientModule after 4.4)
Im fairly new to Angular 5, so it might be my inexperience but i could really use some help.

My **App.module.ts** in library app,
import { BrowserModule } from '@angular/platform-browser';
import { NgModule } from '@angular/core';
import { AppComponent } from './app.component';
import { SpeedometerModule } from './modules/speedometer/speedometer.module';
import { SpeedometerService } from './modules/speedometer/speedometer.service';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    SpeedometerModule,
  ],
  providers: [SpeedometerService],
  bootstrap: [AppComponent]
})
export class AppModule { }

**Speedometer.module.ts** 
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { SpeedometerComponent } from './speedometer.component';
import { HttpClientModule, HttpClient } from '@angular/common/http';
import { SpeedometerService } from './speedometer.service';

@NgModule({
  imports: [
    CommonModule,
    HttpClientModule,
  ],
  declarations:[SpeedometerComponent],
  providers:[SpeedometerService],
  exports:[SpeedometerComponent]
})
export class SpeedometerModule { }

speedometer.service.ts 
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable()
export class SpeedometerService {

  constructor(private http:HttpClient) { }

  test(){
    console.log("Hello World");
  }
}

**ng-package.json**

{
    "$schema": "./node_modules/ng-packagr/ng-package.schema.json",
    "whitelistedNonPeerDependencies": ["."],
    "lib": {
      "entryFile": "public_api.ts",
      "externals": {
      }
    }
}
**angular-cli.json**
..
"defaults": {
    "styleExt": "scss",
    "component": {},
    "build": {
      "preserveSymlinks": true
    }
  }

Does anyone have any suggestions to what i be doing wrong?

#### Version Information
```
Library
ng-packagr: 2.4.1
@angular/*: 1.7.3
typescript: 2.5.3
rxjs: 5.5.6
node: 4.1.0
npm/yarn: 5.6.0

App consuming:
@angular/*: 1.7.4
typescript: 2.4.2
rxjs: 5.5.8
node: 3.2.0
npm/yarn: 5.6.0
_Please include all version numbers that might be relevant, e.g. third-party libraries_

**Comment by chenthillrulz** [comment]
am facing the same issue. any solutions ?

**Comment by Howisch92** [comment]
Well, yes i sort of did. But i still don't know what was wrong. i have an idea, but at this point i cant prove it.

So i build my own component library following this guide (https://medium.com/@nikolasleblanc/building-an-angular-4-component-library-with-the-angular-cli-and-ng-packagr-53b2ade0701e),
and it seemed to be "consumed" just fine, because as long as i didn't have any httpclient dependencies in my services. Note, that the consuming app, was a rather big application, which I've been developing for a few months, and i'm fairly new to angular, so there definitely a lot of things that could have gone wrong.


So i build a new library following dherges sample code, (https://github.com/dherges/ng-packagr/tree/master/integration/samples/core/src) and that resulted in a library with a bit different package hierarchy, i consumed the component library in a new fresh project, with no dependencies, and it worked just perfect.

so what i suspect went wrong was some dependencies in my old(bad) project, overlapping, or pointing differently, than the DI's i had in my component library.

If you're all out of suggestions, i would recommend build a small sample of your current library, following the structure of dherges, and consume the library by a fresh newly made app.

If you figure out the actual problem, i would love to hear from you. I've unfortunately gone to far, that it would require a lot of changes to find out more in detail what was wrong.
And im on a deadline with this project.

Good luck to you, and i hope you might find something here that will help you.

________________________________
Fra: bdonchenko <notifications@github.com>
Sendt: 12. april 2018 16:24:08
Til: dherges/ng-packagr
Cc: Howisch92; State change
Emne: Re: [dherges/ng-packagr] Error: StaticInjectorError[Injector]: when consuming library (#764)


I have a similar issue. did you solve it?

**Comment by chenthillrulz** [comment]
Thanks a lot!! I figured out the issue with mine. I had the shared module in my Main app which was published in npm and was consumed by the feature module which was again getting published in npm.
Main consumed the feature module from npm,  but main app included the shared module from the code base rather than the npm causing different dependency for services. 

So once I figured this and made sure Main app consumes referes the shared services from the npm shared module rather than referencing from its code base, the issue got fixed.
